File: spring-boot/converter/src/main/java/com/example/converter/converter/BooleanToTinyintConverter.java
package com.example.converter.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class BooleanToTinyintConverter implements AttributeConverter<Boolean, Integer> {


    @Override
    public Integer convertToDatabaseColumn(Boolean attribute) {
        return (attribute != null && attribute) ? 1 : 0;
    }

    @Override
    public Boolean convertToEntityAttribute(Integer dbData) {
        return 1 == dbData;
    }
}

File: spring-boot-kotlin/converter/src/main/kotlin/com/example/converter/member/converter/BooleanToTinyintConverter.kt
package com.example.converter.member.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class BooleanToTinyintConverter : AttributeConverter<Boolean, Int?> {
    override fun convertToDatabaseColumn(attribute: Boolean?): Int {
        return if (attribute != null && attribute) 1 else 0
    }

    override fun convertToEntityAttribute(dbData: Int?): Boolean {
        return 1 == dbData
    }
}
